src/coreclr/jit/lower.cpp

```diff
@@ -9119,7 +9134,7 @@ bool Lowering::OptimizeForLdp(GenTreeIndir* ind)
 bool Lowering::TryMakeIndirsAdjacent(GenTreeIndir* prevIndir, GenTreeIndir* indir)
```

Comment:
The `baseAddr` for both `prevIndir` and `indir` is assumed to be same? Can we have an assert for it?

Reply:
It's not that easy to assert (it won't be exactly the same, just something that we expect the emitter peephole to kick in for). But it's also not a precondition for this function that the addresses must be related in some way -- the function works fine even without that. So I don't think there is a good reason to try to assert it.

Reply:
I mean will we mistakenly use offsets of different base address and combine them, leading to bad codegen?

Reply:
This reordering here doesn't combine the stores or loads. It just puts them next to each other. Combining them is done by the peephole in the emitter.

Reply:
Sure, what I meant is we might end up reordering unrelated `indir` and `prevIndir` but the peephole emitter will make sure that we combine the correct matching `indir` and `prevIndir` only.

Reply:
Yes, but this function is still correct even if you ask it to make two indirs off of unrelated addresses adjacent. There is no correctness requirement that the addresses relate in some way; this is not a precondition for the function. Hence why I don't see a good reason to try to assert that (and furthermore, it isn't easy to assert it).

Reply:
> Yes

Ok, just wanted to confirm my understanding.